ecosystem/sep-0012.md

```diff
@@ -363,3 +365,114 @@ PUT [KYC_SERVER || TRANSFER_SERVER]/customer/callback
 | Success                                   | `200 OK`           |
 | Client not authenticated properly         | `401 Unauthorized` |
 | Anchor has no information on the customer | `404 Not Found`    |
+
+## Customer Files
+
+Passing binary fields such as `photo_id_front` or `organization.photo_proof_address` in [PUT /customer](#customer-put) requests must be done using the `multipart/form-data` content type. This is acceptable in most cases, but `multipart/form-data` does not support nested data structures such as arrays or sub-objects.
+
+This endpoint is intended to decouple requests containing binary fields from requests containing nested data structures, supported by content types such as `application/json`. This endpoint is optional and only needs to be supported if your use case requires accepting nested data structures in `PUT /customer` requests.
+
+Once a file has been uploaded using this endpoint, it's `file_id` can be used in subsequent `PUT /customer` requests. The field name for the `file_id` should be the appropriate [SEP-9](sep-0009.md) field followed by `_file_id`. For example, if `file_abc` is returned as a `file_id` from [POST /customer/files](post-request), it can be used in a `PUT /customer` request like so:
+
+```json
+{
+   "account":"GBORFR3GDNVZ5PLUTBDQHKGWVD26CQUHORO2T3SDQ2JPLGLUJCCA5GK6",
+   "memo":"21bf91a4-7db1-401d-8108-fab7660a45d6",
+   "memo_type":"text",
+   "photo_id_front_file_id": "file_abc"
+}
+```
+
+```json
+{
+   "id": "2f417dab-18d2-4081-8c59-c9d3afb59d3f",
+   "photo_id_front_file_id": "file_abc"
+}
+```
+
+### POST Request
+
+```
+POST [KYC_SERVER || TRANSFER_SERVER]/customer/files
+```
+
+Name                 | Type     | Description            
+-------------------- | -------- | ----------------------
+`file` | binary | A file to upload. The file should follow the specifications of [RFC 2388](https://tools.ietf.org/html/rfc2388) (which defines file transfers for the `multipart/form-data` protocol).
+
+### POST Response
+
+Name                 | Type     | Description            
+-------------------- | -------- | ----------------------
+`file_id` | string | Unique identifier for the object.
+`content_type` | string | The `Content-Type` of the file.
+`size` | integer | The size in bytes of the file object.
+`expiration` | [UTC ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) string | (optional) The date and time the file will be discarded by the server if not referenced by the client in a `PUT /customer` request.
```

Comment:
I think we should name this field `expires_at` for similar reasons we discussed in another PR.
```suggestion
`expires_at` | [UTC ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) string | (optional) The date and time the file will be discarded by the server if not referenced by the client in a `PUT /customer` request.
```